multiqc/modules/checkm2/checkm2.py

```diff
@@ -0,0 +1,155 @@
+import logging
+
+from multiqc.base_module import BaseMultiqcModule, ModuleNoSamplesFound
+from multiqc.plots import table
+from multiqc.plots.table_object import TableConfig
+
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    """The module parses the `quality_report.tsv` files generated by CheckM2.
+    All statistics for all samples are saved to `multiqc_data/checkm2-first-table.txt`.
+
+    This has been tested with CheckM2 v1.0.2 .
+    """
+
+    def __init__(self):
+        super(MultiqcModule, self).__init__(
+            name="CheckM2",
+            anchor="checkm2",
+            href="https://github.com/chklovski/CheckM2",
+            info="A rapid, scalable and accurate tool for assessing microbial genome quality using machine learning.",
+            doi=["10.1038/s41592-023-01940-w"],
+        )
+
+        self.checkm2_data = {}
+        for f in self.find_log_files("checkm2", filehandles=True):
+            self.parse_file(f)
+            self.add_data_source(f)
+        self.checkm2_data = self.ignore_samples(self.checkm2_data)
+        if len(self.checkm2_data) == 0:
+            raise ModuleNoSamplesFound
+        log.info(f"Found {len(self.checkm2_data)} reports")
+        self.add_software_version()
+
+        self.mag_quality_table()
+
+    def parse_file(self, f):
+        """Parse the quality_report.tsv output."""
+        column_names = (
+            "Name",
+            "Completeness",
+            "Contamination",
+            "Completeness_Model_Used",
+            "Translation_Table_Used",
+            "Coding_Density",
+            "Contig_N50",
+            "Average_Gene_Length",
+            "Genome_Size",
+            "GC_Content",
+            "Total_Coding_Sequences",
+            "Total_Contigs",
+            "Max_Contig_Length",
+            "Additional_Notes",
+        )
+        parsed_data = {}
+        for line in f["f"]:
+            if line.startswith("Name\t"):
+                continue
+            column_values = line.rstrip().split("\t")
+            column_values = [None if x == "None" else x for x in column_values]
+            parsed_data[column_values[0]] = {k: v for k, v in zip(column_names, column_values) if v}
+        self.checkm2_data = parsed_data
```

Comment:
Should this line get changed to `self.checkm2_data.update(parsed_data)` to cover if there are multiple logs?

Reply:
Would be a good idea!